SmartDeviceLinkService.m

```diff
@@ -0,0 +1,1170 @@
+    formatterShow.locale = self.localization.locale;
+    formatterShow.dateFormat = self.localization[@"weather-alerts.format.date-time.show"];
+    
+    NSDateFormatter *formatterSpeak = [[NSDateFormatter alloc] init];
+    formatterSpeak.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:0];
+    formatterSpeak.locale = self.localization.locale;
+    formatterSpeak.dateFormat = self.localization[@"weather-alerts.format.date-time.speak"];
+    
+    NSSortDescriptor *sorter = [NSSortDescriptor sortDescriptorWithKey:@"dateExpires" ascending:NO];
+    NSArray *unknownSorted = [unknown sortedArrayUsingDescriptors:@[sorter]];
+    NSMutableArray *requests = [NSMutableArray arrayWithCapacity:unknownSorted.count];
+    
+    for (Alert *alert in unknownSorted) {
+        NSString *chunk = [self.localization stringForKey:@"weather-alerts.speak", alert.title, [formatterSpeak stringFromDate:alert.dateExpires]];
+        NSArray *chunks = [SDLTTSChunk textChunksFromString:chunk];
+        
+        // create an alert request
+        SDLAlert *request = [[SDLAlert alloc] initWithAlertText1:alert.title alertText2:[formatterShow stringFromDate:alert.dateExpires] duration:10000];
+        request.ttsChunks = chunks;
+        [requests addObject:request];
+    }
+
+    [self.manager sendSequentialRequests:requests progressHandler:nil completionHandler:nil];
+    [self.currentKnownAlerts unionSet:unknown];
+    
+    MWInfoType infoType = self.currentInfoType;
+    if ([MWInfoTypeWeatherConditions isEqualToString:infoType]) {
+        WeatherConditions *conditions = notification.userInfo[@"weatherConditions"];
+        [self sendWeatherConditions:conditions withSpeak:NO];
+    } else if ([MWInfoTypeDailyForecast isEqualToString:infoType]) {
+        NSArray *forecast = notification.userInfo[@"dailyForecast"];
+        [self sendForecastList:forecast infoType:infoType withSpeak:NO];
+    } else if ([MWInfoTypeHourlyForecast isEqualToString:infoType]) {
+        NSArray *forecast = notification.userInfo[@"hourlyForecast"];
+        [self sendForecastList:forecast infoType:infoType withSpeak:NO];
+    } else if ([MWInfoTypeAlerts isEqualToString:infoType]) {
+        NSArray *alerts = notification.userInfo[@"alerts"];
+        [self sendAlertList:alerts withSpeak:NO];
+    }
+}
+
+
+#pragma mark - RPC Requests
+
+- (void)registerDisplayLayout:(NSString *)layout {
+    SDLSetDisplayLayout *request = [[SDLSetDisplayLayout alloc] initWithLayout:layout];
+    [self.manager sendRequest:request];
+}
+
+- (void)createChangeUnitsInteractionChoiceSet {
+    SDLChoice *metricChoice = [[SDLChoice alloc] initWithId:MWChoiceSetChangeUnitChoiceIdMetric menuName:self.localization[@"choice.units.metric"] vrCommands:@[self.localization[@"vr.metric"]]];
+    SDLChoice *imperialChoice = [[SDLChoice alloc] initWithId:MWChoiceSetChangeUnitChoiceIdImperial menuName:self.localization[@"choice.units.imperial"] vrCommands:self.localization[@"vr.imperial"]];
+
```

Comment:
Getting an `INVALID_DATA` response because the `vrCommands` are not formatted correctly. It should be:

```
SDLChoice *imperialChoice = [[SDLChoice alloc] initWithId:MWChoiceSetChangeUnitChoiceIdImperial menuName:self.localization[@"choice.units.imperial"] vrCommands:@[self.localization[@"vr.imperial"]]];
```